proposals/testing/NNNN-targeted-interoperability-swift-testing-and-xctest.md

```diff
@@ -0,0 +1,296 @@
+- **Warning-only**: Test failures that were previously ignored are reported as
+  runtime warning issues. It also includes runtime warning issues for XCTest API
+  usage in a Swift Testing context. This is for projects which do not want to
+  see new test failures surfaced due to interoperability.
+
+- **Permissive**: This is the default interoperability mode, which surfaces test
+  failures that were previously ignored. It also includes runtime warning issues
+  for XCTest API usage in a Swift Testing context.
+
+- **Strict**: Warning issues included in the permissive mode can be easily
+  overlooked, especially in CI. The strict mode guarantees that no XCTest API
+  usage occurs when running Swift Testing tests by turning those warnings into a
+  `fatalError`.
+
+Configure the interoperability mode when running tests using the
+`SWIFT_TESTING_XCTEST_INTEROP_MODE` environment variable:
+
+| Interop Mode | Issue behaviour across framework boundary                                  | `SWIFT_TESTING_XCTEST_INTEROP_MODE`            |
+| ------------ | -------------------------------------------------------------------------- | ---------------------------------------------- |
+| Warning-only | XCTest API: ⚠️ Runtime Warning Issue. All Issues: ⚠️ Runtime Warning Issue | `warning-only`                                 |
+| Permissive   | XCTest API: ⚠️ Runtime Warning Issue. All Issues: ❌ Test Failure          | `permissive`, or empty value, or invalid value |
+| Strict       | XCTest API: 💥 `fatalError`. Swift Testing API: ❌ Test Failure            | `strict`                                       |
+
+### Phased Rollout
+
+When interoperability is first available, "permissive" will be the default
+interop mode enabled for new projects. In a future release, "strict" will become
+the default interop mode.
+
+## Source compatibility
+
+As the main goal of interoperability is to change behaviour, this proposal will
+lead to situations where previously "passing" test code now starts showing
+failures. We believe this should be a net positive if it can highlight actual
+bugs you would have missed previously.
+
+You can use `SWIFT_TESTING_XCTEST_INTEROP_MODE=warning-only` in the short-term
+to revert any changes to test pass/fail outcomes as a result of
+interoperability.
+
+## Integration with supporting tools
+
+Interoperability will be first available in future toolchain version,
+hypothetically named `6.X`, where permissive interop mode will be enabled for
+projects. After that, a `7.Y` release would make strict interop mode the
+default.
+
+- Swift Package Manager projects: `swift-tools-version` declared in
+  Package.swift will be used to determine interop mode, regardless of the
+  toolchain used to run tests.
+
+- Xcode projects: Installed toolchain version will be used to determine interop
+  mode.
+
+- Any project can use `SWIFT_TESTING_XCTEST_INTEROP_MODE` to override interop
+  mode at runtime, provided they are on toolchain version `6.X` or newer
```

Comment:
```suggestion
  mode at runtime.
```

No need to say "provided they are on X" as it's implied by the developer not having a time machine :D